amethyst_core/src/system_extra.rs

```diff
@@ -0,0 +1,113 @@
+//! Extra system utilities.
+//!
+//! This modules contains an extension trait for the System trait which adds useful transformation
+//! functions.
+
+use shred::{RunningTime, SystemData};
+use specs::prelude::{Read, System};
+
+/// Extra functionality associated systems.
+pub trait SystemExtra {
```

Comment:
Okay, this is the only thing that's debatable IMO:

* should this be in core? What's the reasoning for it?
* should we call it `SystemExtra` (which implies more functionality will go into this)?

Reply:
@torkleyy see my review for the move to amethyst_core.

I'm pretty sure that could even be moved to an external `specs_pausable` crate if they really wanted too. Its generic enough to be that reusable.

Reply:
I'm just don't think this is a core thing. I'll let a third party decide.